tests/helpers/cooper_test_utils.py

```diff
@@ -1,224 +1,251 @@
+            )
+            observed_constraints[self.ineq_constraints] = ineq_state
 
-            cg0_state = cooper.ConstraintState(violation=cg0_surrogate, strict_violation=cg0_violation)
-            cg1_state = cooper.ConstraintState(violation=cg1_surrogate, strict_violation=cg1_violation)
-        else:
-            cg0_state = cooper.ConstraintState(violation=cg0_violation)
-            cg1_state = cooper.ConstraintState(violation=cg1_violation)
+        if self.has_eq_constraint:
+            eq_state = self._compute_constraint_states(
+                x, self.C, self.d, self.C_sur, self.is_eq_sampled, self.eq_observed_constraint_ratio
+            )
+            observed_constraints[self.eq_constraints] = eq_state
 
-        observed_constraints = {self.constraint_0: cg0_state, self.constraint_1: cg1_state}
+        return cooper.CMPState(observed_constraints=observed_constraints)
 
-        return cooper.CMPState(loss=None, observed_constraints=observed_constraints)
-
-    def compute_cmp_state(self, params):
-        """Computes the state of the CMP at the current value of the primal parameters
+    def compute_cmp_state(self, x: torch.Tensor) -> cooper.CMPState:
+        """
+        Computes the state of the CMP at the current value of the primal parameters
         by evaluating the loss and constraints.
         """
-
-        param_x, param_y = params() if callable(params) else params
-
-        loss = param_x**2 + 2 * param_y**2
-
-        cmp_state = cooper.CMPState(loss=loss)
-
-        if self.use_ineq_constraints:
-            violation_cmp_state = self.compute_violations(params=(param_x, param_y))
-            cmp_state.observed_constraints = violation_cmp_state.observed_constraints
-
+        loss = torch.sum(x**2)
+        violation_state = self.compute_violations(x)
+        cmp_state = cooper.CMPState(loss=loss, observed_constraints=violation_state.observed_constraints)
         return cmp_state
 
+    def compute_exact_solution(self):
+        x = cp.Variable(self.num_variables)
+        objective = cp.Minimize(cp.sum_squares(x))
 
-@pytest.fixture(params=[[0.0, -1.0], [0.1, 0.5]])
-def Toy2dCMP_params_init(device, request):
-    return torch.tensor(request.param, device=device)
-
-
-@pytest.fixture(params=[True, False])
-def Toy2dCMP_problem_properties(request, device):
-    use_ineq_constraints = request.param
-    cmp_properties = dict(use_ineq_constraints=use_ineq_constraints)
-
-    if use_ineq_constraints:
-        exact_solution = torch.tensor([2.0 / 3.0, 1.0 / 3.0], device=device)
-    else:
-        exact_solution = torch.tensor([0.0, 0.0], device=device)
-
-    cmp_properties["exact_solution"] = exact_solution
+        constraints = []
+        if self.has_ineq_constraint:
+            constraints.append(self.A.cpu().numpy() @ x <= self.b.cpu().numpy())
+        if self.has_eq_constraint:
+            constraints.append(self.C.cpu().numpy() @ x == self.d.cpu().numpy())
```

Comment:
Do we ever use both? Would the code work then?

Reply:
No. We don't currently have any tests that have both types of constraints. But the cmp should work 🤔